PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -657,7 +636,7 @@ struct HfTrackIndexSkimCreatorTagSelTracks {
     iCut = 5;
     if (statusProng > 0) {
       for (int iCandType = 0; iCandType < CandidateType::NCandidateTypes; ++iCandType) {
-        if (TESTBIT(statusProng, iCandType) && !isSelectedTrackDCA(trackPt, dca, iCandType)) {
+        if (TESTBIT(statusProng, iCandType) && !isSelectedTrackDcaXY(binsPtTrack, &cutsSingleTrack[iCandType], trackPt, dca[0])) {
```

Comment:
```suggestion
        if (TESTBIT(statusProng, iCandType) && !isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack[iCandType], trackPt, dca[0])) {
```
Shouldn't the `&` be removed?

Reply:
Hi @vkucera ! Thanks a lot for your review! Regarding this point, if I remove the `&` I get this error:

`PWGHF/Utils/utilsAnalysis.h:53:25: error: base operand of '->' has non-pointer type 'const o2::framework::LabeledArray<double>'` 

I guess this is because, in the utils function, I am getting the selection information assuming `cuts` to be a pointer as done in the D meson selector. I found the `&` solve the issue, but I don't know if there are other solutions. If you have any suggestions on this just let me know and I will try

Reply:
OK. In the selectors it is `Configurable<LabeledArray>` (which is a pointer) while in the skim creator it is just `<LabeledArray>` which is not a pointer.
Then maybe, you can make it explicit in the function signature: `const T2* const cutsSingleTrack`

Reply:
Hi @vkucera , thanks a lot for the suggestion! Just for my understanding, you are suggesting modifying the definition of the function making `T2` a pointer explicitly without changing the rest, or am I missing something? Because if I try that I get the following error for the Dplus selector (and similarly for the Ds):

`PWGHF/TableProducer/candidateSelectorDplusToPiKPi.cxx:179:29: note:   mismatched types 'const T2*' and 'o2::framework::Configurable<o2::framework::LabeledArray<double> >'`,

which I am not sure how to fix. If you have any further suggestions I will try them out

Reply:
Hi @vkucera, do you have any further suggestions? Thanks!